demosplan/DemosPlanCoreBundle/Command/Addon/AddonInstallFromZipCommand.php

```diff
@@ -220,12 +222,12 @@ private function setPaths(string $path): void
     /**
      * This will try to copy and unzip the Repo if the path is correct and the repo is not already present in the cache.
      */
-    private function copyAndUnzipFileIfNecessary(OutputInterface $output): void
+    private function copyAndUnzipFileIfNecessary(OutputInterface $output, bool $reinstall): void
     {
         $doesFileExist = file_exists($this->zipSourcePath);
         $addonExistsInCache = file_exists($this->zipCachePath);
 
-        if ($doesFileExist && !$addonExistsInCache) {
+        if ($doesFileExist && (!$addonExistsInCache || $reinstall)) {
```

Comment:
sorry, I cannot follow. Could you please split this and use variables for the conditions according to the meaning of the condition?